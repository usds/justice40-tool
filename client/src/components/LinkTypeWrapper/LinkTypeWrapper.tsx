import React from 'react';
import {Link, useIntl} from 'gatsby-plugin-intl';
import {Link as TrussLink} from '@trussworks/react-uswds';

import {IDefineMessage} from '../../data/copy/common';

export interface ILinkTypeWrapper {
    linkText?: string | JSX.Element;
    internal?: boolean;
    url: string | IDefineMessage;
    openUrlNewTab?: boolean;
    className?: string;
    dataCy?: string;
  }

// eslint-disable-next-line valid-jsdoc
/**
 * This function wraps the two types of links we have. Internal links and
 * external links. Internal links should use the Gatsby <Link> component.
 * Eternal links that will open in a new tab will use the Trussworks
 * <Link> component and external links that stay on the page will use the
 * standard <a> tag. This function allows the instance to choose the type of link
 * along with the props necessary to set new tabs, classes.
 *
 * Note - if the link is an external link and will not open in a new
 * browser tab, ensure that hitting the back button works. This has shown to
 * have errors on edge cases (ie, launching the gmail client with mailto links)
 * and it is the recommendation to not have external links open in the same tab.
 *
 * @param props
 * @returns
 */
const LinkTypeWrapper = (props:ILinkTypeWrapper) => {
  const intl = useIntl();
  let {url} = props;

  if (url && typeof url !== `string`) {
    url = intl.formatMessage(url);
  }

  if (props.internal) {
    return (
      <Link
        to={`${url}`}
        className={props.className ? `usa-link ${props.className}` : `usa-link`}
      >
        {props.linkText}
      </Link>
    );
  } else {
    return props.openUrlNewTab ?
    <TrussLink
      variant={'external'}
      className={props.className}
      href={`${url}`}
      target="_blank"
      rel="noreferrer"
      data-cy={props.dataCy ? props.dataCy : ''}
    >
      {props.linkText}
    </TrussLink> :
    <a
      className={props.className}
      href={url}
      data-cy={props.dataCy? props.dataCy : ''}
    >
      {props.linkText}
    </a>;
  }
};

export default LinkTypeWrapper;
