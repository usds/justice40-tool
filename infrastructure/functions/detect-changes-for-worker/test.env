REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1