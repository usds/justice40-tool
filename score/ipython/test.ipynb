{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
