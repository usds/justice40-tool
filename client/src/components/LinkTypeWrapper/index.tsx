import LinkTypeWrapper from './LinkTypeWrapper';

export default LinkTypeWrapper;
