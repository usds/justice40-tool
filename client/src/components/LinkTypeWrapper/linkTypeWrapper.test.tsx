import * as React from 'react';
import {render} from '@testing-library/react';
import LinkTypeWrapper from './index';
import {LocalizedComponent} from '../../test/testHelpers';

describe('testing all link types', () => {
  it('tests internal links', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <LinkTypeWrapper
            linkText={'test link text'}
            internal={true}
            url={'/methodology'}
            openUrlNewTab={false}
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('tests external links same tab', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <LinkTypeWrapper
            linkText={'test link text'}
            internal={false}
            url={'www.usds.gov'}
            openUrlNewTab={true}
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('tests external links new tab', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <LinkTypeWrapper
            linkText={'test link text'}
            internal={false}
            url={'www.usds.gov'}
            openUrlNewTab={false}
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
