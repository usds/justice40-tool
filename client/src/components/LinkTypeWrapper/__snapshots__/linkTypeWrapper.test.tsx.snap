// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing all link types tests external links new tab 1`] = `
<DocumentFragment>
  <a
    data-cy=""
    href="www.usds.gov"
  >
    test link text
  </a>
</DocumentFragment>
`;

exports[`testing all link types tests external links same tab 1`] = `
<DocumentFragment>
  <a
    class="usa-link usa-link--external"
    data-cy=""
    href="www.usds.gov"
    rel="noreferrer"
    target="_blank"
  >
    test link text
  </a>
</DocumentFragment>
`;

exports[`testing all link types tests internal links 1`] = `
<DocumentFragment>
  <a
    class="usa-link"
    href="/en/methodology"
  >
    test link text
  </a>
</DocumentFragment>
`;
